salt/orch/etcd-migrate.sls

```diff
@@ -0,0 +1,107 @@
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+update_grains:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_grains
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+       - salt: update_grains
+
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master', fun='network.ip_addrs', tgt_type='compound') %}
```

Comment:
I'm still not sure I understand why were doing this in a loop like this? Is there a issue caused by stopping the services on all the relevant nodes at once?

Reply:
Yes, as I said before there is reason for this. Master and Worker nodes they have different sets of services. And first we would like to stop `kube-apiservice` first

Reply:
You misunderstood me :) We can still do masters, and then workers after - with different services - however - right row were doing master 1, then master 2, then master 3, then worker 1, then worker 2, then worker 3, , then worker ....

Doing it that was was needed for upgrade to avoid downtime, however, in this case, downtime is unavoidable, and this is simply extending the downtime.